lib/xandra/cluster.ex

```diff
@@ -0,0 +1,131 @@
+defmodule Xandra.Cluster do
+  use GenServer
+
+  @behaviour DBConnection.Pool
+
+  @default_pool_module DBConnection.Connection
+
+  alias __MODULE__.{ControlConnection, StatusChange, Error}
+
+  require Logger
+
+  defstruct [:options, :supervisor, :pool_module, pools: %{}]
```

Comment:
Personally, I would rename `:supervisor` to `:pools_supervisor`. Reads nicer and clearer :)